### PR TITLE
Theme Showcase: Rename Paid themes to Partner themes

### PR DESCRIPTION
**Pull request by taipeicoder**
## Proposed Changes

As discussed in paYJgx-3AU-p2#comment-3740, we are interested in finding a better term for third-party developed themes. Currently, we use "Paid themes", which can be confusing and hard to translate. As replacement, the following terms were suggested:

- Marketplace themes (https://github.com/Automattic/wp-calypso/pull/79351)
- Third Party themes
- Vendor themes
- Partner themes
- Third Party Premium themes
- Premium Party themes

There are many comments in favor of "Partner themes", so [let's go with that](pekYwv-2tC-p2) as the next best thing:

- paYJgx-3AU-p2#comment-3754
- paYJgx-3AU-p2#comment-3793
- paYJgx-3AU-p2#comment-3800
- p1689262800233069/1689200222.553109-slack-C048CUFRGFQ

Theme Showcase:
| Before | After |
| --- | --- |
 | Screenshot 2023-07-26 at 2 44 28 PM |Screenshot 2023-07-26 at 2 43 55 PM|

Theme Detail page (Atomic staging): 
| Before | After |
| --- | --- |
| Screenshot 2023-07-26 at 2 50 37 PM | Screenshot 2023-07-26 at 2 51 05 PM |

## Testing Instructions

* Head to the Theme Showcase.
* Ensure that the pricing filter option Paid is now Partner.
* Ensure that the theme type badge Paid is now Partner.
* Ensure that the theme type badge's tooltip no longer shows Paid theme, but Partner theme.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by fushar** [comment]
Hmm. Even though the change is simple, but I think the consequence is broad (docs, learning materials, etc). Should we post an HE heads-up post first, and check what relevant materials needs to be updated?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8608941

Thank you @taipeicoder for including a screenshot in the description! This is really helpful for our translators.

**Comment by taipeicoder** [comment]
> Hmm. Even though the change is simple, but I think the consequence is broad (docs, learning materials, etc). Should we post an HE heads-up post first, and check what relevant materials needs to be updated?

Right you are 🙂, I've started a thread in p1690359648689289-slack-C048CUFRGFQ

**Comment by mmtr** [comment]
FYI, we're changing the SEO content of the logged-out theme showcase in https://github.com/Automattic/wp-calypso/pull/79845, so we should stay aligned:

- If `#79845` gets merged first, this PR (`#79871`) should be updated to replace "paid" with "partner" in the new SEO content.
- If this PR (`#79871`) gets merged first, `#79845` should be updated to replace "paid" with "partner" in the new SEO content.

cc @Copons @sailerbe

**Comment by taipeicoder** [comment]
> If `#79845` gets merged first, this PR (`#79871`) should be updated to replace "paid" with "partner" in the new SEO content.

Let's go with this 🙂

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8608941).

**Comment by taipeicoder** [comment]
@mmtr  @Copons seems that #79845 is already using `Partner` instead of `Paid`. Does this mean that there is no dependency considerations? If so, then this PR should be ready for merging.

**Comment by Copons** [comment]
> @mmtr @Copons seems that #79845 is already using `Partner` instead of `Paid`. Does this mean that there is no dependency considerations? If so, then this PR should be ready for merging.

@taipeicoder yep! #79845 is pending translations (which will take time, considering how many new strings are introduced), but I don't see any reason to block this PR.

The LOTS will receive the toggle change earlier, but it won't really create any inconsistencies since the SEO copy is currently not considering the pricing tier anyway.

tl;dr :shipit:
